### PR TITLE
Sleepy pen nerf

### PR DESCRIPTION
**Pull request by CydiaLamiales**
## About The Pull Request
Nerfs the sleepy pen by making it single use as well as requiring a two second delay to inject its reagents as well as it making a message that's visible to everyone but the target.

## Why It's Good For The Game
A silent, infinitely reusable tool that instantly injects 45u of reagents and only for 5 TC is simply too good, and has seen plenty of abuse thorough the years. This PR allows it to retain the intended quiet takedown functionality of it, while removing the ability to use it to spam inject people with anything you want.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/110184118/fc71923e-fa70-498c-9b4d-94e6ed219920

</details>

## Changelog
:cl:PowerfulBacon
balance: The sleepy pen now becomes a regular pen after a single use, requires a 0.5 second progress bar to inject its reagents and shows a visible message to everyone except the target.
/:cl:

### CODE REVIEW

**Comment by St0rmC4st3r** [comment]
ided

**Comment by CydiaLamiales** [comment]
> ided

Mind you this code was written by Bacon

**Review by Rukofamicom — CHANGES_REQUESTED** [review]
Non-refillable is a fantastic change, and one that has been a long time coming given the cost and power of the tool.

That said, also making it almost unusable with the wombo combo of an obvious tell and excessively long delay is awful. This makes the item borderline unusable on any remotely aware target.

I would sooner approve a complete removal than a nerf of this level which will result in players being upset that they wasted TC on something they can't even realistically use until after their target has already been incapacitated by something else.

**Comment by PowerfulBacon** [comment]
The message is not visible to the target. Alternatively we could make the message appear to other people after the injection rather than before so that other people can see that you did it

**Comment by CydiaLamiales** [comment]
> The message is not visible to the target. Alternatively we could make the message appear to other people after the injection rather than before so that other people can see that you did it

So, what should it be?

**Comment by PowerfulBacon** [comment]
I'm fine with just going with a message to other people around the target or a delay so short it's hard to react to but requires a target to be still

File: code/modules/paperwork/pen.dm [hunk 1]
```diff
@@ -189,15 +189,24 @@
...
 
-	if(..())
-		if(reagents.total_volume)
-			if(M.reagents)
-				reagents.trans_to(M, reagents.total_volume, transfered_by = user, method = INJECT)
-
+	if(reagents?.total_volume && M.reagents)
+		// Obvious message to other people, so that they can call out suspicious activity.
+		user.visible_message("<span class='warning'>[user] stabs [M] with [src]!</span>", "<span class='notice'>You prepare to engage the sleepy pen's internal mechanism!</span>", vision_distance = 4, ignored_mobs = list(M))
+		if (!do_after(user, 2 SECONDS, M) || !..())
```
Comment (CydiaLamiales) [inline code/modules/paperwork/pen.dm]:
```suggestion
		if (!do_after(user, 5, M) || !..())
```

**Review by Rukofamicom — CHANGES_REQUESTED** [review]
Make the message visible to others after injection (same time as tiny prick) as well and this is good to go.

Half a second is a much more reasonable use time.

**Review by Rukofamicom — APPROVED** [review]
This looks good to me now.

If the usage remains majorly problematic with these nerfs, I would suggest a future PR try a weaker cocktail and if that still fails a total removal of the item.

**Review by Penwin0 — CHANGES_REQUESTED** [review]
Changelog says 2 seconds, which is incorrect

**Comment by CydiaLamiales** [comment]
> Changelog says 2 seconds, which is incorrect

Ah forgot about that, fixed it.

**Review by Penwin0 — APPROVED** [review]

**Comment by Rukofamicom** [comment]
Folks have 24 hours to make a compelling argument for why this shouldn't be nerfed or else I'm merging this tomorrow

File: code/modules/paperwork/pen.dm [hunk 2]
```diff
@@ -189,15 +189,25 @@
...
 
-	if(..())
-		if(reagents.total_volume)
-			if(M.reagents)
-				reagents.trans_to(M, reagents.total_volume, transfered_by = user, method = INJECT)
-
+	if(reagents?.total_volume && M.reagents)
+		// Obvious message to other people, so that they can call out suspicious activity.
+		to_chat(user, "<span class='notice'>You prepare to engage the sleepy pen's internal mechanism!</span>")
+		if (!do_after(user, 5, M) || !..())
```
Comment (PowerfulBacon) [inline code/modules/paperwork/pen.dm]: Use time defines 0.5 seconds

**Comment by Absolucy** [comment]
tbh I dislike it being one-use only.

perhaps a good compromise would be that its reagent capacity is massively reduced after its initial use - maybe like 10u to 15u, and perhaps add a `do_after` on refilling it, so you can't do so in a hectic situation.

also, `vision_distance` would probably be better off being `COMBAT_MESSAGE_RANGE` (3) for consistency's sake?

**Comment by CydiaLamiales** [comment]
> bh I dislike it being one-use only.
> 
> perhaps a good compromise would be that its reagent capacity is massively reduced after its initial use - maybe like 10u to 15u, and perhaps add a `do_after` on refilling it, so you can't do so in a hectic situation.

Or just buy the poisoned knife if you want to have an item that has more than one use.

File: code/modules/paperwork/pen.dm [hunk 3]
```diff
@@ -189,15 +189,25 @@
...
-				reagents.trans_to(M, reagents.total_volume, transfered_by = user, method = INJECT)
-
+	if(reagents?.total_volume && M.reagents)
+		// Obvious message to other people, so that they can call out suspicious activity.
+		to_chat(user, "<span class='notice'>You prepare to engage the sleepy pen's internal mechanism!</span>")
+		if (!do_after(user, 0.5 SECONDS, M) || !..())
+			to_chat(user, "<span class='warning'>You fail to engage the sleepy pen mechanism!</span>")
+			return
+		reagents.trans_to(M, reagents.total_volume, transfered_by = user, method = INJECT)
+		user.visible_message("<span class='warning'>[user] stabs [M] with [src]!</span>", "<span class='notice'>You successfully inject [M] with the pen's contents!</span>", vision_distance = 4, ignored_mobs = list(M))
```
Comment (CydiaLamiales) [inline code/modules/paperwork/pen.dm]:
```suggestion
		user.visible_message("<span class='warning'>[user] stabs [M] with [src]!</span>", "<span class='notice'>You successfully inject [M] with the pen's contents!</span>", vision_distance = COMBAT_MESSAGE_RANGE, ignored_mobs = list(M))
```

**Comment by Absolucy** [comment]
yeah but i feel bad stabbing people with an actual knife :(

**Comment by Rukofamicom** [comment]
> yeah but i feel bad stabbing people with an actual knife :(

Syringes and hypospray are also both an option.

**Review by PowerfulBacon — APPROVED** [review]
Also an indirect nerf to mass-hypnotisation due to this being the most common tool for the job which is good.

**Comment by Isy232** [comment]
I agree with the justification for a nerf, however the current nerf is so massive it makes the sleepy pen all but useless, we're almost at the "24hr" mark so I'll comment my reasoning in a moment

**Comment by CydiaLamiales** [comment]
> Also, what? a major balance PR only gets to sit for 3 days before being speedmerged?

A lot and I mean a LOT of discussion happened about this in the discord beforehand.

**Comment by Isy232** [comment]
> > Also, what? a major balance PR only gets to sit for 3 days before being speedmerged?
> 
> A lot and I mean a LOT of discussion happened about this in the discord beforehand.

There's no thread for it so it wasn't a discussion easily accessible.

**Comment by Isy232** [comment]
I was initially going to complain about removing the refillability, however buffing the poison knife to take that role instead seems fine to me.

Instead I'll complain about the cost. 5 TC for a single stealthy delayed knockout that cant be used in combat is far too expensive. A single use sleepy pen should be 2 or 3 TC,

One injected, the person will notice their stamina draining and can then still move and emote.
If they're smart they will bypass the mute toxin by PDA'ing the HoS.

I've never seen a sleepy pen loaded with the initial mix lead to a successful kidnapping of the target.

I've been injected with the initial sleepy pen mix twice before and both times I was able to escape and then just tell security who I thought did it.

2 or 3 TC is really what a single use of the initial mix is worth.

**Comment by Isy232** [comment]
> You can still swap out the mix before injecting it, mind you.

Damn I didn't know that, I thought the only way to empty was injecting with it.

Fair, its possible to put an uber deathmix in it since 45u is a lot to work with.

**Comment by Rukofamicom** [comment]
> Other than the fact this was almost speedmerged with no publically available discussion.

It has been discussed on many occasions in the past, there was no single central discussion this time, but the stats speak for themselves one way or another:

Alongside the emag, this is one of the most used uplink items by a fairly wide margin. Unlike the emag, this is almost never interactive for others in any way. 

While this PR doesn't really make it any more interactive for the victims, it does put a strict limit on the number of potential victims and removes it as an instant win in active combat where people are already aware that you are out to get them.

**Comment by Rukofamicom** [comment]
Anyhow, the 24 hours is up.

Sleepy pen will carry on as a tool for single-target assassination or incapacitation.

**Pull request merged**